feature/migration/qrcode/src/main/kotlin/app/k9mail/feature/migration/qrcode/domain/usecase/QrCodeResolutionFilter.kt

```diff
@@ -0,0 +1,30 @@
+package app.k9mail.feature.migration.qrcode.domain.usecase
+
+import android.util.Size
+import androidx.camera.core.resolutionselector.ResolutionFilter
+import kotlin.math.abs
+
+internal class QrCodeResolutionFilter : ResolutionFilter {
+    override fun filter(supportedSizes: List<Size>, rotationDegrees: Int): List<Size> {
+        var bestFit = supportedSizes.first()
+        var bestDistance = bestFit.distanceToTargetPixelCount()
+
+        for (size in supportedSizes.asSequence().drop(1)) {
+            val distance = size.distanceToTargetPixelCount()
+            if (distance <= bestDistance) {
+                bestDistance = distance
+                bestFit = size
+            }
+        }
+
+        return listOf(bestFit)
+    }
+
+    private fun Size.distanceToTargetPixelCount(): Int {
+        return abs(TARGET_PIXEL_COUNT - width * height)
+    }
+
+    companion object {
+        private const val TARGET_PIXEL_COUNT = 1920 * 1080
```

Comment:
I don't understand. Configurable how? Allow the user to select a scanning resolution? I don't think we want to do that.

Reply:
Not by the user, but based on the available camer resolution.

Reply:
A low resolution could mean large QR codes can't be scanned because individual data bits can't be resolved. A higher resolution means higher memory usage and longer image processing times, potentially leading to a janky preview. 
I had a hard time finding data on what resolution to choose. One website (couldn't find it again) mentioned using more than 2 megapixels doesn't significantly improve results. And using this approach worked well enough in my tests, even with QR codes containing a lot of data.